keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -0,0 +1,422 @@
+
+#### As support engineer
+
+- I want to access detailed build information to determine if a customer is running a known problematic version of a component
+
+### Risks and Mitigations
+
+1. **No sensitive data exposed**
+    
+  We will apply the  principle of least privilege by granting only the minimum necessary permissions for accessing /statusz, which is primarily a debugging tool for operators and maintainers.
+  /statusz will not be accessible through general discovery mechanisms like system:discovery. This prevents unintended exposure of potentially sensitive debugging information.
+  
+  We'll leverage Kubernetes RBAC to grant access to /statusz only to specific users and service accounts that require it for operational monitoring and debugging. For this, we will use the existing system:monitoring group ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), ensuring that only authorized personnel can access this endpoint.
+  
+1. **Performance impact**
+    
+  We will take care to carefully consider the computational overhead associated with the statusz endpoint to avoid negatively impacting the server's overall performance.
+
+2. **Essential data only**
+    
+  We will prioritize inclusion of only essential diagnostic and monitoring data that aligns with the intended purpose of the z-page.
+
+3. **Premature Dependency on Unstable Format**
+
+  The alpha release will explicitly support plain text format only, making it clear that the output is not intended for parsing or automated monitoring. The feature will be secured behind a feature gate that is disabled by default, ensuring users opt-in consciously. Also, the endpoint will be implemented with versioned URLs to facilitate future transitions to structured formats without breaking existing integrations, once the usage patterns and requirements are better understood
+
+## Design Details
+
+### Data Format and versioning
+
+Initially, the statusz page will exclusively support a plain text format for responses. We will implement these endpoints using versioned URLs, like /statusz/v1, to ensure future compatibility. This versioning strategy allows us to seamlessly introduce structured data formats (e.g., JSON) in the future, through a distinct endpoint such as /statusz/v2, without disrupting existing implementations.
+
+### Authz and authn
+
+Access to the endpoint will be limited to members of the `system:monitoring group` ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), which is how paths like /healthz, /livez, /readyz are restricted today. 
+
+### Endpoint Response Format
+
+#### Data format : text
+
+#### Request
+* Method: **GET** 
+* Endpoint: **/statusz/v1**
+* Body: empty
+
+#### Response fields
```

Comment:
I think JSON is a saner first format, so we can describe it more precisely.

Reply:
I think we decided (ref: [this](https://docs.google.com/document/d/1BlmHq5uPyBUDlppYqAAzslVbAO8hilgjqZUTaNXUhKM/edit?tab=t.0#bookmark=id.98qzwvyy3b3a) discussion) to start with plain text because we didnt want users to start scraping this data (at least in alpha) with the expectation that the endpoint is stable.

Reply:
You clearly haven't met our users :)  -  parsing text is not an impediment to their willingess to do whatever needs to be done to get the info they need.

Reply:
we can re-hash this. But I think the discussion was to avoid any suggestions that there is a schema. I actually would love to have something more standardized here in future. So limiting us to specific JSON format today would not be ideal.

Reply:
+1 to above, having structured response can probably be interpreted to mean that the endpoint is stable, having plain-text may still serve as a signal that these endpoints are currently only human readable... 

But even then, if users still choose to rely on this, but we would have provided clear indications (also somewhere clearly stated in the endpoint's response itself) that it's not recommended.

Reply:
That's fine, just don't be surprised when you find people parsing anyway :)